### PR TITLE
Around FlyTo Hash / User

### PR DESCRIPTION
**Pull request by fuentesloic**

### CODE REVIEW

File: static/js/around-map.js [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
   center: [-10, 45],
   zoom: 2
 })
-
 map.addControl(new mapboxgl.NavigationControl())
 map.addControl(new mapboxgl.GeolocateControl())
 
-// users interesting points
+// detectionLocation
+if (currentLocation.length == 0) {
```
Comment (vinyll) [inline static/js/around-map.js]: remove the space before the opening parenthesis

File: static/js/around-map.js [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
+  focusUser()
+} else if (currentLocation.length > 0) {
```
Comment (vinyll) [inline static/js/around-map.js]: remove the space before the opening parenthesis

File: static/js/around-map.js [hunk 3]
```diff
@@ -29,55 +39,87 @@ worker.onmessage = response => {
       .setLngLat([poi.position.longitude, poi.position.latitude])
       .setPopup(popup)
       .addTo(map)
+
+    if (window.location.hash.indexOf('#') == 0) {
```
Comment (vinyll) [inline static/js/around-map.js]: Why not just `if(window.location.hash)` ?
Remove the space before the opening parenthesis

File: static/js/around-map.js [hunk 4]
```diff
@@ -29,55 +39,87 @@ worker.onmessage = response => {
...
+      const hash = getHash()
+      if (hash == el.id) {
```
Comment (vinyll) [inline static/js/around-map.js]: remove space

File: static/js/around-map.js [hunk 5]
```diff
@@ -29,55 +39,87 @@ worker.onmessage = response => {
...
+      bearing: 0,
+      speed: 1.7,
+      curve: 1
+    })
+  }
+})
 
-worker.postMessage('info-requested')
+// functions
+const params =
```
Comment (vinyll) [inline static/js/around-map.js]: Hum… ???

**Review by vinyll — CHANGES_REQUESTED** [review]
You could refactor
```
map.flyTo({
    center: currentLocation,
    zoom: 11,
    bearing: 0,
    speed: 1.7,
    curve: 1
  })
```

to a function like `moveMapTo(position)` with:

```
function moveMapTo (position) {
map.flyTo({
    center: position,
    zoom: 11,
    bearing: 0,
    speed: 1.7,
    curve: 1
  })
}
```

File: static/js/around-map.js (hunk 5 above)
Comment (fuentesloic) [reply static/js/around-map.js]: sorry I forget I was trying something ...

**Comment by fuentesloic** [comment]
Ok for the refactoring !! i would do that but I don't find out how :')

**Comment by vinyll** [comment]
>Ok for the refactoring !! i would do that but I don't find out how :')
❓

**Comment by vinyll** [comment]
And please test in your browser with your browser console when you push a PR so that mistakes like `const params =` don't remain, and we review working code ;)

**Comment by fuentesloic** [comment]
I can't refactor with this one 
`
map.flyTo({
      center: [poi.position.longitude, poi.position.latitude],
      zoom: 11,
      bearing: 0,
      speed: 1.7,
      curve: 1
    })
`

**Comment by vinyll** [comment]
Then you can refactor with the others, right?
Why can't you refactor that one?

**Comment by fuentesloic** [comment]
I was writting how I can't refactor all of them and during it I find ... I'll try

**Comment by vinyll** [comment]
ok, just try to make a:

```
function moveMap (position, somethingelse) {
  map.flyTo({stuff})
}
```

and play with that.
You should be able to do something with less repeat ;)

File: static/js/around-map.js [hunk 6]
```diff
@@ -1,21 +1,31 @@
...
 })
-
 map.addControl(new mapboxgl.NavigationControl())
 map.addControl(new mapboxgl.GeolocateControl())
 
-// users interesting points
+// detectionLocation
+if(currentLocation.length == 0) {
+  focusUser()
+} else if(currentLocation.length > 0) {
```
Comment (vinyll) [inline static/js/around-map.js]: replace your `else if(currentLocation.length > 0) {` with a `else {`

File: static/js/around-map.js [hunk 7]
```diff
@@ -29,55 +39,73 @@ worker.onmessage = response => {
...
-  if(currentLocation.length > 0) {
-    currentMarker(currentLocation.reverse())
-  } else {
-    navigator.geolocation.getCurrentPosition(position => {
-      currentMarker([position.coords.longitude, position.coords.latitude])
-    })
-  }
+function focusUser () {
+  navigator.geolocation.getCurrentPosition(position => {
+  const position = [position.coords.longitude, position.coords.latitude]
```
Comment (vinyll) [inline static/js/around-map.js]: indent this block

**Review by vinyll — CHANGES_REQUESTED** [review]

**Comment by fuentesloic** [comment]
what do you think now ?

**Comment by fuentesloic** [comment]
what do you think now ? (I didn't do that you said ...)

File: static/js/around-map.js [hunk 8]
```diff
@@ -82,9 +80,8 @@ function currentMarker (currentLocation) {
 
 function focusUser () {
   navigator.geolocation.getCurrentPosition(position => {
-  const position = [position.coords.longitude, position.coords.latitude]
-  moveTo(position)
-  currentMarker(position)
+    moveTo([position.coords.longitude, position.coords.latitude])
+    currentMarker(location)
```
Comment (vinyll) [inline static/js/around-map.js]: What is `location`?

File: static/js/around-map.js [hunk 9]
```diff
@@ -81,7 +81,7 @@ function currentMarker (currentLocation) {
 function focusUser () {
   navigator.geolocation.getCurrentPosition(position => {
     moveTo([position.coords.longitude, position.coords.latitude])
-    currentMarker(location)
+    currentMarker([position.coords.longitude, position.coords.latitude])
```
Comment (vinyll) [inline static/js/around-map.js]: If you use twice the same array, you may want to store it in a variable ;)

Comment (vinyll) [reply static/js/around-map.js]: And use it in both function

**Review by vinyll — CHANGES_REQUESTED** [review]

File: static/js/around-map.js (hunk 8 above)
Comment (fuentesloic) [reply static/js/around-map.js]: that was an oversight

File: static/js/around-map.js [hunk 10]
```diff
@@ -80,8 +80,9 @@ function currentMarker (currentLocation) {
 
 function focusUser () {
   navigator.geolocation.getCurrentPosition(position => {
-    moveTo([position.coords.longitude, position.coords.latitude])
-    currentMarker([position.coords.longitude, position.coords.latitude])
+    userPosition = [position.coords.longitude, position.coords.latitude]
```
Comment (vinyll) [inline static/js/around-map.js]: missing a `const`

**Review by vinyll — CHANGES_REQUESTED** [review]

**Review by vinyll — APPROVED** [review]
Cool! You just need to squash your commits ;)

**Comment by vinyll** [comment]
Hum, no… 12 commits??? 😱 
For 1 functionality in 1 file

**Comment by fuentesloic** [comment]
we just addd one functionality: move according the hash

**Comment by vinyll** [comment]
Right. I'm just confused as the PR is mentioning 12 commits and the actual master bran is just one commit with all the comments (which is less of a drama ;) )

**Pull request merged**